### ISSUE TITLE
Openshift.process Signature

### ISSUE THREAD

**Issue by mmkonrad** [openshift/jenkins-client-plugin#166]
Hello together,

due to reasons the assembly of our required parameters for the `openshift.process` is a little bit complicated an I have a question about that.

Your HOWTO says:
> // One straightforward way is to pass string arguments directly to `oc process`.
// This includes any parameter values you want to specify.
def models = openshift.process( "openshift//mongodb-ephemeral", "-p", "MEMORY_LIMIT=600Mi" )

Our call for `openshift.process` is located within a method that has several parameters:
- a string with the template name
- a map containing generated parameters for the template
- sometimes a name for a parameter file that contains static parameters for a specific environment (file gets unstashed upfront)

Until recently whe had a parameterized method call, similar to your example:
```            
def entities = openshift.process(
            '-f', "${templateFileName}", 
            '-p', "ANOTHER_PARAMETER_NAME=${anotherValue}", 
            '-p',  ANOTHER_PARAMETER_NAME=${anotherValue}", 
            '-p', "ANOTHER_PARAMETER_NAME=${anotherValue}")
```
But it became to inflexible. We have the need for a varying parameter signature.

My question is: How do I put them together accordingly, so that `openshift.process` works as intended?

[Your code](https://github.com/openshift/jenkins-client-plugin/blob/master/src/main/resources/com/openshift/jenkins/plugins/OpenShiftDSL.groovy) beginning in Line does not give that away for me. (I am not a java/groovy professional and jenkins pipelines work even more differently)

The attempt with an array that contains the parameters in correct order does not work (unfirtunately we do not receive some kind of error message from the plugin):

```
            def argsArray = []
         
            argsArray.push("-f")
            argsArray.push("${templateFileName}")

            if (propertyFileName != null) {
                argsArray.push('--param-file')
                argsArray.push(propertyFileName)
            }

            templateArgs.each {
                argsArray.push('-p')
                argsArray.push("${it}")
            }

           def entities = openshift.process(argsArray)
```
The `argsArray` contains all our required parameters in correct order and all of them are of `class java.lang.String`. 

Where am I wrong and how ca I fix it?

**Comment by gabemontero** [comment]
Hi @mmkonrad 

Rather than employing `-f, ${templateFileName}`, first use the jenkins pipeline  [(readFile step)](https://jenkins.io/doc/pipeline/steps/workflow-basic-steps/#readfile-read-file-from-workspace) to read in the json content into string variable, then pass that json variable into `openshift.serializableMap` to convert the json string into a Map variable.

Then call 
`openshift.process(mapVar, variableSizeArgumentArray)`

I believe variableSizeArgumentArray can either be a Object[] or  List.

Bottom line, groovy makes is difficult to support all the `oc process` permutations along with a variable argument list var arrays/lists.

**Comment by mmkonrad** [comment]
Thank you for your reply.

Luckily I was able to fix it myself by sticking to the initial proposal of your README.
A simple concatenation of a string did the trick and works just fine:

```
            String args = ""
            args = "${args} -f ${templateFileName}"
        
            if (propertyFileName != null) {
                args = "${args} --param-file ${propertyFileName}"
            }

            templateArgs.each {
                args = "${args} -p ${it}"
            }
```
In case further adjustments are required I will have your proposal in mind.

Thank you again
